artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -543,6 +602,17 @@ public void preAcknowledge(final Transaction tx, final MessageReference ref, fin
          logger.trace("preAcknowledge::tx={}, ref={}, reason={}", tx, ref, reason);
       }
 
+      SimpleString noForwardSource = null;
+      if (Boolean.TRUE.equals(ref.getMessage().getBooleanProperty(INTERNAL_NO_FORWARD))) {
+         noForwardSource = (SimpleString) ref.getMessage().getBrokerProperty(INTERNAL_NO_FORWARD_SOURCE);
```

Comment:
I dont think we should be setting a SimpleString value for this...and now looking at the other uses, it doesnt seem like it will be, its set as a String originally? (Which makes sense as its ultimately coming from an AMQP link property, as a string value).

Reply:
Even though the value we set on the target is a String, it gets somehow turned into a SimpleString. If I try to cast that broker property to a string, I'm getting: `class org.apache.activemq.artemis.api.core.SimpleString cannot be cast to class java.lang.String`.

On the target side I do send a String in the property:
![image](https://github.com/user-attachments/assets/78522d37-78b4-4a8b-a87f-5ca1dc736da4)

Reply:
Seems to be the fault of setBrokerProperty() using TypedProperties for 'extra properties' (as well as setting a message annotation if AMQP message), which converts a String value to a SimpleString :/

I'd check the exact behaviour first, but I'd be inclined to flip the behaviour so it checks with a String instead, i.e that it null checks then does String.valueOf to get a String for the comparison. That way its not creating new SimpleString from the [typically same] remoteMirrorId value for every message checked, but instead just getting the String from the SimpleString (that likely already exists because it was  created from a String).

Reply:
Flipping the behavior works for me, but depending on the volume of changes that may or may not require, I think this could endup being off-topic for the PR.

Reply:
When I speak of flipping the behaviour I just mean the comparison done within the if statements here, not anything else. Thats definitely on-topic for this PR given it is adding the bit that it would be good not to do.

Reply:
Okay makes sense, that was some misunderstanding on my side, sorry about that.